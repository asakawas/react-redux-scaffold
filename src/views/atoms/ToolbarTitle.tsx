import Typography from '@material-ui/core/Typography';
import styled from 'styled-components';

const ToolbarTitle = styled(Typography)`
  flex-grow: 1;
`;

export default ToolbarTitle;
