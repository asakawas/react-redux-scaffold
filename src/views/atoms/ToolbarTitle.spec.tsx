import React from 'react';
import { render } from '@testing-library/react';
import ToolbarTitle from './ToolbarTitle';

describe('ToolbarTitle', () => {
  test('is rendered as expected', () => {
    const { asFragment } = render(<ToolbarTitle />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('has "flex-grow: 1" in its style attribute', () => {
    const renderResult = render(<ToolbarTitle data-testid="test" />);

    const tree = renderResult.getByTestId('test');
    expect(tree).toHaveStyleRule('flex-grow', '1');
  });
});
