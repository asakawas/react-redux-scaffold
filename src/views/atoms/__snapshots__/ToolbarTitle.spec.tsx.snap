// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolbarTitle is rendered as expected 1`] = `
<DocumentFragment>
  .c0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<p
    class="MuiTypography-root c0 MuiTypography-body1"
  />
</DocumentFragment>
`;
